### PR TITLE
Remove redundant `len` and `nil` check before loop

### PR DESCRIPTION
**Pull request by Juneezee**
## What does this PR change?

From the Go specification (https://go.dev/ref/spec#For_range):

> "1. For a nil slice, the number of iterations is 0."
> "3. If the map is nil, the number of iterations is 0."

`len` returns 0 if the slice or map is nil (https://pkg.go.dev/builtin#len). Therefore, having these checks before a loop is unnecessary:

- `len(v) != 0`
- `len(v) > 0`
- `v != nil`

## Does this PR relate to any other PRs?
No.

## How will this PR impact users?
No user-facing change.

## How was this PR tested?
`go build ./...` and `go test ./...`

## Does this PR require changes to documentation?
No.

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next OpenCost release? If not, why not?
No. External contributor cannot label PR.

### CODE REVIEW

**Review by Juneezee — COMMENTED** [review]
Tip: Hide the whitespace to make the diff easier to read

image

image

**Comment by AjayTripathy** [comment]
Thank you for this PR. I suspect these hooks were left in the code to quickly add an 

else {
  // log error
}

since many of these seem to imply exceptional cases except for this https://github.com/opencost/opencost/pull/2184/files#diff-084f229ce009968d50ab3e5cbeb8ff6f42eb0fd08fdc7b9d027f6931d47ea483R1765

**Comment by Juneezee** [comment]
@cliffcolvin Friendly ping :smiley:

**Comment by AjayTripathy** [comment]
Hey @Juneezee I think we'd rather log something about these values being nil in an else clause, as I mentioned earlier, than drop the if altogether since these cases feel exceptional.

File: pkg/cloud/aws/provider.go
Comment (Juneezee) [inline pkg/cloud/aws/provider.go:1]: This one is redundant. I don't think we need a `log`.

File: pkg/costmodel/costmodel.go
Comment (Juneezee) [inline pkg/costmodel/costmodel.go:1]: This one is redundant. I don't think we need a `log`.

File: pkg/costmodel/metrics.go
Comment (Juneezee) [inline pkg/costmodel/metrics.go:1]: This one is redundant. I don't think we need a `log`.

**Review by Juneezee — COMMENTED** [review]
@AjayTripathy Updated. Please take a look again. Thanks!

**Comment by mattray** [comment]
@AjayTripathy do you want to take a quick look?

**Pull request closed without merging**